### PR TITLE
Use xhr-request module for much smaller builds

### LINKED ISSUE #34: Propose replacing request module with something smaller for browser usage
The `request` module is reeeally large and builds are >1MB. This isn't such a big deal when using with node, but is not so good for browser usage.

I would suggest maybe [xhr-request](https://github.com/Jam3/xhr-request) instead. 

More info in this thread. https://twitter.com/nickpeihl/status/796806578305253376

I will start working on a PR.

### PR DESCRIPTION
**Pull request by nickpeihl**
[xhr-request](https://github.com/Jam3/xhr-request) is a much smaller alternative to the standard request library. This reduces the build size from 1.1MB to 36KB! 

xhr-request works well for both browsers and node with very few differences from the request library.

Feel free to squash the commits before merging. Fixes #34.

### CODE REVIEW

**Comment by jgravois** [comment]
very cool!

**Comment by dmfenton** [comment]
Sorry I missed this until now!

File: index.js [hunk 1]
```diff
@@ -85,20 +77,23 @@ FeatureService.prototype.request = function (url, callback) {
...
+    headers: {
+      'user-agent': 'Featureservices-Node'
+    }
+
   }
-  this._request(options, function (err, res) {
+  this._request(encoded, options, function (err, data, res) {
     if (err) {
-      if (err.message === 'ESOCKETTIMEDOUT') err.code = 504
+      if (err.code === 'ESOCKETTIMEDOUT') err.code = 504
```
Comment (dmfenton) [inline index.js]: why is this changed? There are some pretty specific needs upstream in Koop that depend on the error structure.
